### ISSUE TITLE
[Bug] Job resources are not getting created with terraform helm_release

### ISSUE THREAD

**Issue by mohamedazouz** [temporalio/helm-charts#404]
### What are you really trying to do?

I'm trying to install helm chart using [terraform helm provider](https://registry.terraform.io/providers/hashicorp/helm/latest/docs) for that I'm using  terraform `helm_release` resource to install the chart. 

I have created a ticket in [terraform-helm-provider](https://github.com/hashicorp/terraform-provider-helm/issues/1122) repo but it seems like the issue exists in chart itself.

### Describe the bug

I'm using the following terraform code to install the chart 

```hcl
// Temporal Chart: https://github.com/temporalio/helm-charts

resource "helm_release" "chart" {
  name             = "temporaltest"
  chart            = /local/Temporal/chart/path"
  create_namespace = true
  values           = []

  dependency_update = true
  timeout           = 15 * 60
}

```
After applying above terraform code, all pods are created but they are pending for init containers to finish, it's being pending forever, here's result of `kubectl get pods`

```
temporal-cassandra-0                  1/1     Running             0          2m2s
temporal-history-6f985967d8-wrngs     0/1     Init:2/4            0          2m6s
temporal-frontend-85f6dbdd85-44s4s    0/1     Init:2/4            0          2m6s
temporal-matching-65c75675d8-lfwz5    0/1     Init:2/4            0          2m6s
temporal-worker-5c46bd7dcc-2bhf2      0/1     Init:2/4            0          2m8s
```

#### Observation
* Jobs resources are not getting created when using `helm_release` unlike when we use helm install
* We tried to use `helm_template` to get `manifest` output for the chart then deploy it using `kubectl apply`, it worked as expected

#### Expected Behavior
* All resources in the chart should be created similarly like when I use helm install:
```bash
> helm install  temporaltest . --timeout 15m
```

### Minimal Reproduction

1. `git clone https://github.com/temporalio/helm-charts`
2. Adjust local chart path to the cloned repo path
3. `terraform apply`

### Environment/Versions

```
Terraform version: v1.2.0
Provider version: 2.9.0
Kubernetes version: 1.23
```

**Comment by pmorillas-apheris** [comment]
Hey @mohamedazouz 

Run into the same issue and setting the `wait = false` should workaround it
https://github.com/hashicorp/terraform-provider-helm/issues/683

**Comment by mohamedazouz** [comment]
@pmorillas-apheris sorry for the late reply.

I'm not sure if you understand the issue i'm facing but [wait](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#wait) will ignore the not-yet-created resources.
The issue is that all pods are in init state and waiting for Cassandra schema setup job to create the proper schema but unfortunately, jobs are not being created for some reason which is which i'm asking for help.

Your workaround of using `wait = false `, it assumes that the issue is just wait but it's not it's because jobs are not started to setup database schema,

**Comment by LukaGiorgadze** [comment]
@mohamedazouz Getting the same issue. I'm working now to find out how to solve it.

**Comment by robholland** [comment]
This should now be fixed by #522 which removes the use of helm hooks entirely. Please re-open if you still see problems.
